### DISCUSSION TITLE
Simple example of isotropic rotor blade section for CAMRAD use

### DISCUSSION THREAD

**Discussion by rhig2000** [wenbinyugroup/.github#12] [VABS Related Discussions] [answered]
For the provided example, angle values are the same for both principal inertial and principal bending axes: 0.135250640821118 degrees about the positive direction of x1 axis. Also, the mass, geometric, and tension centers of the cross section are all the same: 5.3088098940E-01, 5.6705323980E-03. Therefore, how can expected CAMRAD offsets XI, ZI, and XC, ZC, be different?
XI= 0.0727908
ZI= -0.00327108
XC= 0.0725671
ZC= 0.00657381

**Comment by rhig2000** [comment]
P.S. Additional related question: Is 6X6 Mass Matrix the correct data source to use for resolving principal inertial axes?

CAMRAD variables for pitch angle of principal inertial and principal bending axes are THETAI and THETAC. Eq. 52 from "Inertial and Structural Properties of General Composite Beams" by Wenbin Yu, computes the same value as VABS 4.1 for THETAC from Timoshenko Compliance Matrix, but Eq. 26 does not for THETAI from 6X6 Mass Matrix.

**Reply by wenbinyu-PU** [reply]
> P.S. Additional related question: Is 6X6 Mass Matrix the correct data source to use for resolving principal inertial axes?
> 
> CAMRAD variables for pitch angle of principal inertial and principal bending axes are THETAI and THETAC. Eq. 52 from "Inertial and Structural Properties of General Composite Beams" by Wenbin Yu, computes the same value as VABS 4.1 for THETAC from Timoshenko Compliance Matrix, but Eq. 26 does not for THETAI from 6X6 Mass Matrix.

The reason being you need the 6x6 mass matrix measured with respect to elastic axis (or shear center). Note the primes for quantities needed in Eq. (26).

**Reply by rhig2000** [reply]
Ok. I used Eq. 15 to compute primes from 6X6 Mass Matrix for Eq. 26. Is 'The Shear Center of the Cross Section in the User Coordinate System' the correct data source for shear center offsets?

**Reply by rhig2000** [reply]
Ok. my values from Eq.15 going into Eq. 26 are:
1. ip22 = 0.00011276519012107209
2. ip23 = -9.968377695685599e-05
3. ip33 = 0.01879798128294964

So, H = 0.009343139834355115

Are these correct?

**Reply by wenbinyu-PU** [reply]
Yes, correct.

**Reply by rhig2000** [reply]
Great! Thanks for confirming that!
So, using python function for theta = math.asin(- ip23 / H) / 2, I get 0.005334698506975149 radians = 0.3056557094244176 degrees. Where am I going wrong?

**Reply by wenbinyu-PU** [reply]
Thanks a lot for all the tough questions. These numbers are correct. The corresponding XI=0.0728072 and ZI=-0.00288272. I need to correct my previous answer. In fact, XI is not the same as XC and ZI is not the same as ZC. Because the principal bending axes is different from the principal inertial axes. The reason being the principal bending axes do not change with respect to origin, while the principal inertial axes change with respect to the origin. These two sets of axes are the same at the mass center/geometry center/tension center.

**Comment by wenbinyu-PU** [comment, marked as answer]
> For the provided example, angle values are the same for both principal inertial and principal bending axes: 0.135250640821118 degrees about the positive direction of x1 axis. Also, the mass, geometric, and tension centers of the cross section are all the same: 5.3088098940E-01, 5.6705323980E-03. Therefore, how can expected CAMRAD offsets XI, ZI, and XC, ZC, be different? XI= 0.0727908 ZI= -0.00327108 XC= 0.0725671 ZC= 0.00657381

Corrected: The main reason here is what I said in another reply: the principal bending axes is different from the principal inertial axes. The reason being the principal bending axes do not change with respect to origin, while the principal inertial axes change with respect to the origin. These two sets of axes are the same at the mass center/geometry center/tension center.

**Reply by rhig2000** [reply]
My formulas compute XI=XC= 0.0725671 ZI=ZC= 0.00657381 for θ = 0.13525064 radians, not degrees. Are the wrong angle units stated in the VABS output file or are XI, ZI, and XC, ZC values incorrect?

**Reply by rhig2000** [reply]
I struggling to reconcile "the principal bending axes is different from the principal inertial axes" with angle values given in the VABS output file, which are the same for both principal inertial and principal bending axes: 0.135250640821118 degrees about the positive direction of x1 axis. Could you provide a diagram?

**Reply by wenbinyu-PU** [reply]
![image](https://github.com/user-attachments/assets/549451f3-a97a-42c9-8890-c85c503ebf4b)

**Reply by rhig2000** [reply]
What are all the angles, generally, and their correct values for the example problem?

**Reply by wenbinyu-PU** [reply]
Only two different angles: 
the first angle is 0.13524 degrees (for principal bending/inertial axes at mass/geometry/tension center, and for principal bending axes at shear center).
the second angle is 0.305656 degrees (for principal inertial axes as shear center).

**Reply by rhig2000** [reply]
Oh, wow! 💡Principal inertial axes change with respect to the origin? I was not expecting that. So, Eq. 26 calculates principal inertial axes at **shear** center, but from main_cs_set_iso.sg.K, the following refers to principal inertial axes at **mass** center:

>  The Mass Properties with respect to Principal Inertial Axes
>  ========================================================
>  
>  Mass per unit span                     =    2.4633502327E-01
>  Mass moment of inertia i11             =    1.7602903596E-02
>  Principal mass moments of inertia i22  =    1.1003255936E-04
>  Principal mass moments of inertia i33  =    1.7492871036E-02
>  The principal inertial axes rotated from user coordinate system by 
>   0.135240607448158      degrees about the positive direction of x1 axis.
>  The mass-weighted radius of gyration   =    2.6731853583E-01

Since principal inertial axes change with respect to the origin, please consider indicating the origin for mass properties in VABS output.

**Reply by wenbinyu-PU** [reply]
Great suggestion. I am changing it now.

**Reply by rhig2000** [reply]
I appreciate your support very much!
Evidently CAMRAD requires principal inertial axes at shear center for THETAI, or beam axis per 'Volume VI: Rotorcraft Input', Figure 20-3, p. 214

**Reply by wenbinyu-PU** [reply]
So you answered your own question.

**Comment by wenbinyu-PU** [comment]
Yes.
